pkg/controller/instanceset/instance_util.go

```diff
@@ -216,38 +220,67 @@ func buildInstanceName2TemplateMap(itsExt *instanceSetExt) (map[string]*instance
 	return allNameTemplateMap, nil
 }
 
-func GenerateAllInstanceNames(parentName string, replicas int32, templates []InstanceTemplate, offlineInstances []string) []string {
+func GenerateAllInstanceNames(parentName string, replicas int32, templates []InstanceTemplate, offlineInstances []string, templatesIndexRanges []workloads.InstanceTemplateIndexRanges) ([]string, error) {
```

Comment:
it is better to add a new function and only scope in instance controller.

Reply:
That sounds good, thank you.